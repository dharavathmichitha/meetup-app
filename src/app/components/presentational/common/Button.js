import React from 'react';
import PropTypes from 'prop-types';

const Button = ({ cssClass, text, clickHandler }) => {
  return (
    <button type="button" className={`${cssClass} btn`} onClick={clickHandler}>
      {text}
    </button>
  );
};

Button.propTypes = {
  cssClass: PropTypes.string.isRequired,
  text: PropTypes.string.isRequired,
  clickHandler: PropTypes.func.isRequired,
};
export default Button;
