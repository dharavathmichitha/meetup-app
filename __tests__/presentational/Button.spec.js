import React from 'react';
import renderer from 'react-test-renderer';
import Button from '../../src/app/components/presentational/common/Button';

beforeEach(() => {
  jest.resetModules();
});

describe('Button Component', () => {
  it('should match snapshot including default props', () => {
    const props = {
      cssClass: 'event__sing-up',
      clickHandler: () => true,
    };
    const component = renderer.create(<Button {...props}>test text</Button>);

    const componentTree = component.toJSON();

    expect(componentTree).toMatchSnapshot();
  });
});
