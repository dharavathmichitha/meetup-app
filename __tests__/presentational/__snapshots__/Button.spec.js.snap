// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Component should match snapshot including default props 1`] = `
<button
  className="event__sing-up btn"
  data-hover="Cancel"
  onClick={[Function]}
  type="button"
>
  test text
</button>
`;
